packages/actor-abstract-mediatype-utilities/lib/AbstractMediatypeUtilities.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  DataSources,
+  IDataSource,
+  KEY_CONTEXT_SOURCE,
+  KEY_CONTEXT_SOURCES,
+} from "@comunica/bus-rdf-resolve-quad-pattern";
+import {ActionContext} from "@comunica/core";
+
+/**
+ * A comunica Hash Query Operation Actor.
+ */
+export abstract class AbstractMediatypeUtilities {
+
+  /**
+   * Get the single source if the context contains just a single source.
+   * @param {ActionContext} context A context, can be null.
+   * @return {Promise<IDataSource>} A promise resolving to the single datasource or null.
+   */
+  public static async getSingleSource(context: ActionContext): Promise<IDataSource> {
+    if (context && context.has(KEY_CONTEXT_SOURCE)) {
+      // If the single source is set
+      return context.get(KEY_CONTEXT_SOURCE);
+    } else if (context && context.has(KEY_CONTEXT_SOURCES)) {
```

Comment:
This block should not be needed, this will always be handled by the federated actor. You can assume that the context will always have a `KEY_CONTEXT_SOURCE` entry.